connect_deprish_to_ferc1.py

```diff
@@ -682,14 +740,79 @@ def split_ferc1_data_on_split_cols(self,
 
         # merge in the newly generated split/assigned data column
         df_final = pd.merge(
-            same_smol, df_w_tots[
-                idx_cols_ferc1 + ['record_id_eia_deprish', new_data_col]])
+            same_smol,
+            df_w_tots[idx_cols_ferc1 + ['record_id_eia_deprish',
+                                        new_data_col]].drop_duplicates(),
+        )
         return df_final
 
+    def agg_ferc_data_cols(self):
+        """
+        Aggregate smaller level FERC1 data columns to depreciation records.
+
+        When the depreciation matches are at a higher level than the FERC1,
+        this method aggregates the many FERC1 records into the level of the
+        depreciation records.
+        """
+        data_cols_own_frac = [
+            f"{col}_own_frac" for col in DATA_COLS_TO_SPLIT.keys()]
```

Comment:
Pretty sure you don't need the `.keys()` here. Iterating over a dictionary iterates over its keys.